spacelift/resource_drift_detection.go

```diff
@@ -0,0 +1,167 @@
+package spacelift
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shurcooL/graphql"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+)
+
+func resourceDriftDetection() *schema.Resource {
+	return &schema.Resource{
+		Description: "" +
+			"`spacelift_drift_detection` represents a Drift Detection configuration " +
+			"for a Stack. It will run a proposed run on the given schedule, which you can " +
+			"listen on using run state webhooks. If reconcile is true, then a tracked run " +
+			"will be triggered when drift is detected.",
+
+		CreateContext: resourceDriftDetectionCreate,
+		ReadContext:   resourceDriftDetectionRead,
+		UpdateContext: resourceDriftDetectionUpdate,
+		DeleteContext: resourceDriftDetectionDelete,
+
+		Importer: &schema.ResourceImporter{StateContext: importIntegration},
+
+		Schema: map[string]*schema.Schema{
+			"reconcile": {
+				Type:        schema.TypeBool,
+				Description: "Whether a tracked run should be triggered when drift is detected.",
+				Optional:    true,
+			},
+			"stack_id": {
+				Type:        schema.TypeString,
+				Description: "ID of the stack for which to set up drift detection",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"schedule": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				MinItems:    1,
+				Description: "List of cron schedule expressions based on which drift detection should be triggered.",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceDriftDetectionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var mutation struct {
+		CreateDriftDetectionIntegration struct {
+			Reconcile bool     `graphql:"reconcile"`
+			Schedule  []string `graphql:"schedule"`
+		} `graphql:"stackIntegrationDriftDetectionCreate(stack: $stack, input: $input)"`
+	}
+
+	var scheduleExpressions []graphql.String
+	for _, expr := range d.Get("schedule").([]interface{}) {
+		scheduleExpressions = append(scheduleExpressions, graphql.String(expr.(string)))
+	}
+
+	stackID := d.Get("stack_id").(string)
+
+	variables := map[string]interface{}{
+		"stack": toID(stackID),
+		"input": structs.DriftDetectionIntegrationInput{
+			Reconcile: graphql.Boolean(d.Get("reconcile").(bool)),
+			Schedule:  scheduleExpressions,
+		},
+	}
+
+	if err := meta.(*internal.Client).Mutate(ctx, "DriftDetectionUpdate", &mutation, variables); err != nil {
+		return diag.Errorf("could not create drift detection integration for the stack: %v", err)
```

Comment:
Terraform's error message will not include the stack name if this is shown, which will make things harder to debug especially when creating multiple drift configurations at once.

Please consider including the stack name in error messages about sub-stack resources. And in general showing the parent resource ID in child resource error messages. 🙏